### PR TITLE
Add SBOM Generation to V1

### PR DESCRIPTION
**Pull request by michaelpeng36**
### Issue describing the changes in this PR

Adds the SBOM generation task to the build pipeline for Core Tools V1 and moves the pipeline over from AppVeyor to Azure DevOps.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by Francisco-Gamino — CHANGES_REQUESTED** [review]
The code change looks good. I left some comments. Let me know if you have any questions.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+variables:
+  devops_buildNumber: $[counter(format(''), 1500)]
+  APPVEYOR_REPO_BRANCH: $[coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranchName'])]
+  APPVEYOR_REPO_COMMIT: $(Build.SourceVersion)
+
+steps:
+- pwsh: |
+    $isReleaseBuild = $false
+    if ($env:BuildSourceBranchName -like "release_1.0*")
```
[the remark this answers is not in the document]
Comment (michaelpeng36) [reply azure-pipelines.yml:26]: As it stands now, none of the development branches or release branches contain a `/`, and I do not see a case in the near future where that will be necessary. I will keep this in mind in the case that it does, however.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    - v1.x
+
+pool:
+  name: '1ES-Hosted-AzFunc'
+  demands:
+  - ImageOverride -equals MMS2022TLSSDK8
+
+variables:
+  devops_buildNumber: $[counter(format(''), 1500)]
+  APPVEYOR_REPO_BRANCH: $[coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranchName'])]
```
[the remark this answers is not in the document]
Comment (michaelpeng36) [reply azure-pipelines.yml:20]: Yes, they should. These changes will need to be made for the other branches as well.

File: pipelineUtilities.psm1 [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                & .\$installScript -InstallDir "$env:ProgramFiles/dotnet" -Channel $Channel -Version $Version
+            } else {
+                bash ./$installScript --install-dir /usr/share/dotnet -c $Channel -v $Version
+            }
+        }
+        AddLocalDotnetDirPath
+    }
+    finally {
+        Remove-Item $installScript -Force -ErrorAction SilentlyContinue
+    }
```
Comment (davidmrdavid) [inline pipelineUtilities.psm1:112]: Do we have any error handling in case this `Install-Dotnet` step fails?

**Review by davidmrdavid — COMMENTED** [review]
Left a minor, non-merge-blocking, question. In my opinion, after renaming the appveyor variables in the `yml`, I think this should be ready to merge.

File: pipelineUtilities.psm1 (hunk 3 above)
Comment (michaelpeng36) [reply pipelineUtilities.psm1:112]: As it currently stands, no, but my understanding is that if this step fails, the installation of dotnet has gone sufficiently wrong to warrant throwing an exception.

File: build.fsx [hunk 4]
```diff
@@ -31,23 +31,29 @@ let MoveFileTo (source, destination) =
...
-let deployDir = "./deploy" + platform + "/"
+let artifactsDir = "./artifacts/"
+let telemetryDir = artifactsDir + "SBOMManifestTelemetry/"
 let downloadDir  = "./dist/download" + platform + "/"
+let manifestToolDll = Path.GetFullPath (manifestToolDir @@ "Microsoft.ManifestTool.dll")
+let packageName = "Azure.Functions.Cli." + platform
 let sigCheckExe = toolsDir @@ "sigcheck.exe"
+let dotnetExe = "dotnet.exe"
 let nugetUri = Uri ("https://dist.nuget.org/win-x86-commandline/v3.5.0/nuget.exe")
 let version = if isNull appVeyorBuildVersion then "1.0.0.0" else appVeyorBuildVersion
```
Comment (Francisco-Gamino) [inline build.fsx:48]: Could you please rename these variable `appVeyorBuildVersion` and `appVeyorBuildVersion` as well? Thanks.

**Review by Francisco-Gamino — APPROVED** [review]
LGTM

**Review by davidmrdavid — APPROVED** [review]
LGTM!

File: build.fsx (hunk 4 above)
Comment (Francisco-Gamino) [reply build.fsx:51]: Thank you for adding a comment.

**Review by Francisco-Gamino — APPROVED** [review]
Thank you for completing this work. LGTM!

**Pull request merged**